internal/brokers/withexamples.go

```diff
@@ -38,7 +38,7 @@ func useExampleBrokers() (string, func(), error) {
 	return cfgPath, func() {
 		conn.Close()
 		if err := os.RemoveAll(cfgPath); err != nil {
-			log.Warningf(context.Background(), "Failed to remove the broker configuration directory: %v", err)
+			slog.Warn(fmt.Sprintf("Failed to remove the broker configuration directory: %v", err))
```

Comment:
Isn't `slog.Warn` supporting formatting natively?

Reply:
Mh, that's a bit annoying because in some code paths where we'd end up logging a lot (and potentially with complex parameters = e.g. PAM) it implies that we have to do everywhere `slog.Default().Enabled()` before calling the function or we'd end up allocating a lot of strings (sometimes causing introspection recursions in objects) for no reasons.

Reply:
One thing I want to note is that I don't think it's worth to put a lot of effort into optimizing the code that handles a login request, because those are initiated by the user, so they are relatively rare, and it doesn't matter much if it takes a few milliseconds longer to respond do it. That said, if the PAM module really logs so many complex parameters that it notably affects performance, we can consider wrapping the `slog` package with a package that provides `Warnf`, `Infof`, etc. functions and only does the string formatting if the log level is enabled.

I would like to postpone that until we actually use `slog` in the PAM module, which is not in scope of this PR.

Reply:
> That said, if the PAM module really logs so many complex parameters that it notably affects performance, we can consider wrapping the slog package with a package that provides Warnf, Infof, etc. functions and only does the string formatting if the log level is enabled.

Well it's mostly when used in gdm, where its polling nature makes it to waste a lot of resources already, so trying to optimize it's something we should try as much as possible